src/lrtd.rs

```diff
@@ -0,0 +1,256 @@
+//! Utility to help with "really nice to add a warning for tasks that might be blocking"
+use libc;
+use rand::thread_rng;
+use rand::Rng;
+use std::collections::HashSet;
+use std::sync::mpsc;
+use std::sync::{Arc, Mutex};
+use std::time::Duration;
+use std::{env, thread};
+use tokio::runtime::{Builder, Runtime};
+
+const PANIC_WORKER_BLOCK_DURATION_DEFAULT: Duration = Duration::from_secs(60);
+
+fn get_panic_worker_block_duration() -> Duration {
+    let duration_str = env::var("MY_DURATION_ENV").unwrap_or_else(|_| "60".to_string());
+    duration_str
+        .parse::<u64>()
+        .map(Duration::from_secs)
+        .unwrap_or(PANIC_WORKER_BLOCK_DURATION_DEFAULT)
+}
+
+fn get_thread_id() -> libc::pthread_t {
+    unsafe { libc::pthread_self() }
+}
+
+/// A trait for handling actions when blocking is detected.
+///
+/// This trait provides a method for handling the detection of a blocking action.
+pub trait BlockingActionHandler: Send + Sync {
```

Comment:
take it or leave it: it might be nice to add an `impl BlockingActionHandler for F where F: Fn(&[libc::pthread_t])`?